src/math/tri.js

```diff
@@ -0,0 +1,73 @@
+import { Vec3 } from "./vec3.js";
+
+/** @typedef {import('../shape/ray.js').Ray} Ray */
+
+// Internal variables
+const h = new Vec3();
+const s = new Vec3();
+const q = new Vec3();
+
+/**
+ * Class for representing a triangle in 3D space
+ */
+class Tri {
+    /**
+     * Creates a new Triangle Object.
+     *
+     * @param {Vec3} vertex0 - Vertex 0 value
+     * @param {Vec3} vertex1 - Vertex 1 value
+     * @param {Vec3} vertex2 - Vertex 2 value
+     * @param {Vec3} [centroid] - Centroid value
+     */
+
+    constructor(vertex0, vertex1, vertex2, centroid) {
```

Comment:
I think you only call it from one place, without centroid parameter .. is it still required as a parameter to the constructor?

Reply:
The centroid parameter is no longer in the constructor, and is initialized as a Vec3. New function `calculateCentroid()` has been added to the class